CMake/FindWolfSSL.cmake

```diff
@@ -0,0 +1,9 @@
+find_path(WOLFSSL_INCLUDE_DIRS wolfssl/wolfssl.h)
+
+find_library(WOLFSSL_LIBRARY wolfssl)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(WOLFSSL DEFAULT_MSG
+    WOLFSSL_INCLUDE_DIRS WOLFSSL_LIBRARY)
+
+mark_as_advanced(WOLFSSL_INCLUDE_DIRS WOLFSSL_LIBRARY)
```

Comment:
@bagder the case of the first argument to the `find_package_handle_standard_args` call should match the case of the find module.  That's `WolfSSL` in this case.

The name of the header file should be the same as on `#include` lines.  For curl I see plain `#include "wolfssl.h"` lines so the `wolfssl/` prefix should not be used here because we want the `-I` flag to point at a directory containing the header directly.

Also please take a look at the [cmake-developer(7)](https://cmake.org/cmake/help/v3.17/manual/cmake-developer.7.html) manual section on writing find modules.  It documents conventions for variable names and such.  You should use a singular name for the individual `find_*` result variables, e.g. `find_path(WolfSSL_INCLUDE_DIR ...)`.  Those are the variables that should be listed in the `find_package_handle_standard_args` call as required variables because those names are reported in error messages to the user as missing.

So something like:

```cmake
find_path(WolfSSL_INCLUDE_DIR NAMES wolfssl.h)
find_library(WolfSSL_LIBRARY NAMES wolfssl)
mark_as_advanced(WolfSSL_INCLUDE_DIR WolfSSL_LIBRARY)

include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(WolfSSL
  REQUIRED_VARS WolfSSL_INCLUDE_DIR WolfSSL_LIBRARY
  )

if(WolfSSL_FOUND)
  set(WolfSSL_INCLUDE_DIRS ${WolfSSL_INCLUDE_DIR})
  set(WolfSSL_LIBRARIES ${WolfSSL_LIBRARY})
endif()
```

The plural-named variables set at the end are the public results that the rest of the code outside this find module should use.

Reply:
Awesome, thanks @bradking !

We include `<wolfssl/ssl.h>` in the code so I've adjusted the test accordingly. Will push the update in a sec.

How do I pass in a custom directory prefix for cmake to look for wolfssl in? I have wolfssl installed in `$HOME/build-wolfssl`

Reply:
With CMake 3.12 or higher you can pass `-DWolfSSL_ROOT=...` to point at that directory.  Or you can set the same variable/value pair in your environment.  This will work since curl already has the line

```cmake
cmake_minimum_required(VERSION 3.0...3.16 FATAL_ERROR)
```

The `...3.16` part will cause policy [CMP0074](https://cmake.org/cmake/help/v3.17/policy/CMP0074.html) to be set to `NEW`, thus enabling support for `<PackageName>_ROOT` variables on `find_package` calls.

With CMake versions older than that (or newer) one can pass the directory in [`CMAKE_PREFIX_PATH`](https://cmake.org/cmake/help/v3.17/variable/CMAKE_PREFIX_PATH.html) via `-DCMAKE_PREFIX_PATH=$HOME/build-wolfssl`.  If more than one such location is needed they can be in a `;`-separated list in the same variable.

Reply:
Thanks!